src/network/server.rs

```diff
@@ -302,98 +307,109 @@ impl MyHubService {
+            .map_err(|_| HubError::internal_db_error("stores not found for fid"))?;
+
+        let id_register = stores
+            .onchain_event_store
+            .get_id_register_event_by_fid(fid)
+            .map_err(|_| HubError::internal_db_error("Could not fetch id registration"))?;
+
+        match id_register {
+            None => return Err(HubError::validation_failure("missing fid registration")),
+            Some(id_register) => {
+                match id_register.body {
+                    Some(Body::IdRegisterEventBody(id_register)) => {
+                        // Check verified addresses if the resolved address doesn't match the custody address
+                        if id_register.to != resolved_ens_address {
+                            let verification = VerificationStore::get_verification_add(
+                                &stores.verification_store,
+                                fid,
+                                &resolved_ens_address,
+                            )?;
+
+                            match verification {
+                                None => Err(HubError::validation_failure("invalid ens proof, no matching custody address or verified addresses")),
+                                Some(_) => Ok(()),
+                            }
+                        } else {
+                            Ok(())
+                        }
+                    }
+                    _ => return Err(HubError::validation_failure("missing fid registration")),
+                }
             }
-            Some(l1_client) => {
-                let name = std::str::from_utf8(&proof.name)
-                    .map_err(|_| HubError::validation_failure("ENS name is not utf8"))?;
+        }
+    }
 
+    async fn resolve_ens_address(&self, proof: &UserNameProof) -> Result<Vec<u8>, HubError> {
+        let name = std::str::from_utf8(&proof.name)
+            .map_err(|_| HubError::validation_failure("ENS name is not utf8"))?;
+
+        let chain_api = match UserNameType::try_from(proof.r#type) {
+            Ok(UserNameType::UsernameTypeEnsL1) => {
                 if !name.ends_with(".eth") {
                     return Err(HubError::validation_failure(
                         "ENS name does not end with .eth",
                     ));
                 }
-
-                let resolved_ens_address = l1_client
-                    .resolve_ens_name(name.to_string())
-                    .await
-                    .map_err(|err| {
-                        HubError::validation_failure(
-                            format!("ENS resolution error: {}", err.to_string()).as_str(),
-                        )
-                    })?
-                    .to_vec();
-
-                if resolved_ens_address != proof.owner {
+                self.chain_clients.for_chain(Chain::EthMainnet)?
+            }
+            Ok(UserNameType::UsernameTypeBasename) => {
+                if !name.ends_with(".eth") {
```

Comment:
.base.eth?

Reply:
Fixed. We can probably get rid of this check though, by the time we get here message validation would've already ensured this is correct.